### DISCUSSION TITLE
3,5Inch RPI display 480 320 XPT2046 touch not working

### DISCUSSION THREAD

**Discussion by jnivard** [Bodmer/TFT_eSPI#2362] [Q&A - Touch] [answered]
Hai 

I try to get a display working using a ESP32 Wroom32 DevKit V1. The display is working ok but the touch is not reacting
To get started I used the instructions from jarutex at :   https://www.jarutex.com/index.php/2021/12/24/8919/  using the following setup: (see below).

When I run the TouchTest program I recieve the following output: x:2352 y:3347 z 5025 and the values change a litle bit, but do not react on a touch of the screen.  Demo_3D_cube is working fine

#define RPI_DISPLAY_TYPE ok
#define ILI9486_DRIVER   ok
#define TOUCH_CS        18       ok                      Chip select Touch      
#define TFT_MISO 2  ok
#define TFT_MOSI 12 ok
#define TFT_SCLK 14 ok                                     clock signal
#define TFT_CS 5 ok                                            Chip select signal TFT
#define TFT_DC 15  
#define TFT_RST 13 

#define LOAD_GLCD 
#define LOAD_FONT2 
#define LOAD_FONT4 
#define LOAD_FONT6 
#define LOAD_FONT7 
#define LOAD_FONT8 
#define LOAD_GFXFF 
#define SMOOTH_FONT 
#define SPI_FREQUENCY 26000000
#define SPI_TOUCH_FREQUENCY 2500000

I have checked the wiring and have a collor coding for each wire and that seems not the problem. I did buy the display from aliexpress. It looks like the display of https://www.waveshare.com/3.5inch-RPi-LCD-A.htm, but it is some clone. 
3.5 Inch Tft Lcd Contact Screen 320X480 Resolutie Abs Case + Touchpen Voor Raspberry Pi 4 B  https://nl.aliexpress.com/item/1005003983760114.html?spm=a2g0o.order_list.order_list_main.17.5b2279d2eSeiKH&gatewayAdapt=glo2nld 

I like to know How I can proceed from here

**Comment by Bodmer** [comment, marked as answer]
Try moving the MISO signal to a different pin, say pin 4. Pin 2 is connected to an LED.

Try dropping the touch SPI frequency to 1000000

**Reply by jnivard** [reply]
Hai Thanks for your reply

It is not helping, I reduced the SPI frequency and moved the pin 2 to 4 but it has the only effect that the demo of the cube is moving slowly. It has no effect on the TouchTest the readings are in the range of x:2352 y:3347 z 5025 and touching the screen has no effect. I lowerd also the frequency SPI_TOUCH_FREQUENCY but it has no effect. I did buy two boards and used thy other board and now it is working ok, many thanks for your excelend work. I will try to make a PCB board for the PY case using the ESP32 development board and will chare the result with you.

Many thanks John
